shmem.c

```diff
@@ -105,18 +105,18 @@ void newOverTimeClient() {
 }
 
 void addOverTimeClientSlot() {
-	// For each client slot, add an overTime slot
-	for(int i = 0; i < overTimeClientCount; i++) {
-		// Get the name of the shared memory
-		char *name = clientShmName(i);
+	// For each client slot, add pagesize overTime slots
+	for(int i = 0; i < overTimeClientCount; i++)
+	{
+		// Only increase size of the shm object is needed
+		if(shm_overTimeClients[i].size > (size_t)counters->overTime)
+			continue;
 
 		// Reallocate with one more slot
-		realloc_shm(&shm_overTimeClients[i], shm_overTimeClients[i].size + sizeof(int));
+		realloc_shm(&shm_overTimeClients[i], (counters->overTime/pagesize + 1)*pagesize*sizeof(int));
```

Comment:
Uses the same math as above. This line is skipped if the object doesn't need to be resized.

Reply:
It might be clearer to use this expression instead: `shm_overTimeClients[i].size + pagesize * sizeof(int)`